go/vt/sqlparser/ast_format.go

```diff
@@ -2760,9 +2718,15 @@ func (node *Variable) Format(buf *TrackedBuffer) {
 	case VariableScope:
 		buf.literal("@")
 	case SessionScope:
+		if node.Name.EqualString("transaction_isolation") || node.Name.EqualString("transaction_read_only") {
+			buf.astPrintf(node, "@@%s.", node.Scope.ToString())
+			break
+		}
 		buf.literal("@@")
 	case GlobalScope, PersistSysScope, PersistOnlySysScope:
 		buf.astPrintf(node, "@@%s.", node.Scope.ToString())
+	case NoScope:
+		buf.literal("@@")
```

Comment:
This doesn't seem correct to me. From MySQL docs - 
```
You can set transaction characteristics globally, for the current session, or for the next transaction only:

With the GLOBAL keyword:

The statement applies globally for all subsequent sessions.

Existing sessions are unaffected.

With the SESSION keyword:

The statement applies to all subsequent transactions performed within the current session.

The statement is permitted within transactions, but does not affect the current ongoing transaction.

If executed between transactions, the statement overrides any preceding statement that sets the next-transaction value of the named characteristics.

Without any SESSION or GLOBAL keyword:

The statement applies only to the next single transaction performed within the session.

Subsequent transactions revert to using the session value of the named characteristics.
```

So if no scope is provided, we shouldn't change it to session scope

Reply:
I see that we disallow no scope in vtgate later, but even then, we `sqlparser` output should be correct in the sense that the scope of the query isn't changed implicitly. This is because `sqlparser` as a package is used outside Vitess too

Reply:
I have to add a nextTxScope to explicitly call that out. I just wanted to you have a quick look that it is not insane to change setTransaction to setExpr type